luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/LuthierIndustrialPark.java

```diff
@@ -212,8 +205,28 @@ public GranularityParser getGranularityParser() {
      * @return the LogicalMetric instance corresponding to this metricName.
      */
     public LogicalMetric getMetric(String metricName) {
-        // TODO: to be finished in the metric PR.
-        return null;
+        Map<String, LogicalMetric> metricDictionary = resourceDictionaries.getMetricDictionary();
+        if (metricDictionary.get(metricName) == null) {
+            LogicalMetric metric = metricFactoryPark.buildEntity(metricName, this);
+            metricDictionary.put(metricName, metric);
+        }
+        return metricDictionary.get(metricName);
+    }
+
+    /**
+     * Retrieve or build a Metric Maker.
+     *
+     * @param metricMakerName the name for the dimension to be provided.
```

Comment:
`dimension` should be `metricMakerName`.

Reply:
Good catch